### ISSUE TITLE
#miSelectedItem is used differently in propagate and tags.

### ISSUE THREAD

**Issue by ClotildeToullec** [moosetechnology/MooseIDE#1163]
`#miSelectedItem` is called in the browser to propagate and in the model to tag (`MiAbstractModel>>#tagSelectedEntities`).

We should choose only one way and stick to it.

**Comment by NicolasAnquetil** [comment]
Not sure.
By "used in the model to tag", you mean use the tagging feature of all tools ?

If yes, then this feature is a shortcut to: "propagate miSelectedItems ; go in TagBrowser ; receive entities ; tag them"
So in this sense, the shortcut should apply to the miSelectedItems of the current tool

Actually the 'inspect' button of all tools should also use this miSelectedItems, for the same reasons

The idea is: We "select" some entities in a tool and we can do several actions on them from there: propagate/inspect/tag/may-be-something-else-in-the-futur

**Comment by ClotildeToullec** [comment]
No, what I meant is what you meant in the issue mentioned.
#miSelectedItem is implemented in the Browser and in the BrowserModel.

`MiSomeBrowser>>#miSelectedItem` is called by the Propagate and Inspect mechanisms.
`MiSomeBrowserModel>>#miSelectedItem` is called by the 'Tagging all selected entities' mechanism.

That is weird and may be a source of bugs or inconsistencies.

**Comment by NicolasAnquetil** [comment]
Ok, so if I understand well, and for a nice MVC:
- the `MiAbstractBrowser` should forward the propagate/inspect/tagging requests to its `MiAbstractModel`
-  `MiAbstractModel` should use `#miSelectedItem` to execute the requests

**Comment by ClotildeToullec** [comment]
Yes. I think some (if not most) of the browsers kind of already do this, but each in its own way, so there is a standardization work to do.

**Comment by NicolasAnquetil** [comment]
Agreed 
Excellent initiative ! Go for it!
